test/exchanges/oasis/Eth2DaiDirect.spec.js

```diff
@@ -0,0 +1,223 @@
+      .getContractByName('MAKER_OTC').address;
+    const registryAddress = service
+      .get('smartContract')
+      .getContractByName('PROXY_REGISTRY').address;
+    const daiAddress = service
+      .get('token')
+      .getToken('DAI')
+      .address();
+    const wethAddress = service
+      .get('token')
+      .getToken('WETH')
+      .address();
+    const normalParams = await service._buildParams(
+      'DAI',
+      '0.01',
+      'WETH',
+      service._valueForContract(0, 'WETH'),
+      'sellAllAmount'
+    );
+    const ethParams = await service._buildParams(
+      'WETH',
+      '0.01',
+      'DAI',
+      service._valueForContract(100, 'WETH'),
+      'sellAllAmountPayEth'
+    );
+    const buyAndPayEthParams = await service._buildParams(
+      'WETH',
+      '0.01',
+      'DAI',
+      service._valueForContract(100, 'WETH'),
+      'buyAllAmountPayEth'
+    );
+    const createParams = await service._buildParams(
+      'WETH',
+      '0.01',
+      'DAI',
+      service._valueForContract(100, 'WETH'),
+      'createAndSellAllAmountPayEth'
+    );
+
+    expect(normalParams.length).toEqual(5);
+    expect(normalParams[0]).toEqual(otcAddress);
+    expect(normalParams[1]).toEqual(daiAddress);
+    expect(Object.keys(normalParams[2])).toEqual(['_bn']);
+    expect(normalParams[3]).toEqual(wethAddress);
+    expect(Object.keys(normalParams[4])).toEqual(['_bn']);
+
+    expect(ethParams.length).toEqual(4);
+    expect(ethParams[0]).toEqual(otcAddress);
+    expect(ethParams[1]).toEqual(wethAddress);
+    expect(ethParams[2]).toEqual(daiAddress);
+    expect(Object.keys(ethParams[3])).toEqual(['_bn']);
+
+    expect(buyAndPayEthParams.length).toEqual(4);
+    expect(buyAndPayEthParams[0]).toEqual(otcAddress);
+    expect(buyAndPayEthParams[1]).toEqual(daiAddress);
+    expect(Object.keys(buyAndPayEthParams[2])).toEqual(['_bn']);
+    expect(buyAndPayEthParams[3]).toEqual(wethAddress);
+
+    expect(createParams.length).toEqual(4);
+    expect(createParams[0]).toEqual(registryAddress);
+    expect(createParams[1]).toEqual(otcAddress);
+    expect(createParams[2]).toEqual(daiAddress);
+    expect(Object.keys(createParams[3])).toEqual(['_bn']);
+  });
+
+  test('set transaction options', () => {
+    const normalOptions = service._buildOptions(1, 'DAI', 'method');
+    const ethOptions = service._buildOptions(1, 'ETH', 'method');
+    const createOptions = service._buildOptions(1, 'ETH', 'create');
+    expect(Object.keys(normalOptions)).toEqual(['otc', 'dsProxy']);
+    expect(Object.keys(ethOptions)).toEqual(['otc', 'dsProxy', 'value']);
+    expect(Object.keys(createOptions)).toEqual(['otc', 'value']);
+  });
+});
+
+describe('values from otc', () => {
+  beforeAll(async () => {
+    await createDaiAndPlaceLimitOrder(service);
+  });
+
+  test('get buy amount', async () => {
+    const buyAmount = await service.getBuyAmount('WETH', 'DAI', '0.01');
+    expect(Object.keys(buyAmount)).toEqual(['_bn']);
```

Comment:
there's no need to check this, because the expectation below will also fail if this isn't true